enhancements/rhcos/split-rhcos-into-layers.md

```diff
@@ -0,0 +1,428 @@
+
+* As an OpenShift customer, I want to be able to follow (most of) the image mode for RHEL documentation and use (most of) the same tooling surrounding it when customizing RHCOS, so that I have a consistent experience across image mode for RHEL and RHCOS and a clear understanding of how they relate to each other.
+
+* As an ART engineer, I want to be able to build the OpenShift node image in a similar way to how the other OpenShift components are built, so that I can simplify my tooling and pipelines.
+
+* As a RHEL kernel engineer, I want to be able to debug an OCP kernel bug the same way I do for image mode for RHEL, so that I don't have to learn different ways to do things.
+
+* As an OKD contributor, I want to be able to use CentOS Stream-based boot images, so that I don't have to keep pivoting from Fedora CoreOS boot images into CentOS Stream everytime.
+
+* As an OKD contributor, I want to be able to share node image building processes with OCP, so that I can reduce the maintenance load.
+
+* (Longer-term) As a product manager, I want image mode for RHEL certification and support guarantees to automatically apply to OpenShift clusters.
+
+### Goals
+
+- Build RHCOS/SCOS on top of rhel-bootc/centos-bootc
+- Share efforts, especially CI and docs, with the larger rhel/centos-bootc project
+- Maintain a single RHCOS stream per RHEL stream
+- Build the OpenShift node image as a layered image on top of RHCOS/SCOS
+- Build the OpenShift node image using Prow (in CI) and Konflux (in prod)
+
+### Non-Goals
+
+- Change the cluster installation flow. It should remain the same whether IPI/UPI/AI/etc...
+- Introduce cluster administrator-visible changes. This change should be transparent to
+  admistrators. CoreOS layering instructions should keep working as is, but documentation
+  should ideally be reworked to leverage rhel-bootc docs more.
+- Remove Ignition. While image mode for RHEL lessens the need for Ignition, it remains a
+  crucial part of the provisioning flow and will stick around for the foreseeable future.
+
+## Proposal
+
+The overall proposal consists of two parts. In the first part, we will focus on moving
+the OCP-specific packages out of RHCOS and into its own layer (the node layer). In the second
+part, we'll change RHCOS itself to become a layered image (the CoreOS layer) built on top
+of the rhel-bootc work (the bootc layer).
+
+As mentioned in the summary, this enhancement is focused on detailing the first part only.
+Firstly, because it's the one we'll start with, and secondly, because it's the one that
+will most impact OpenShift. The second part will require changes that are more contained
+to the CoreOS and ART teams. That said, we felt it important to introduce them together
+in this enhancement to provide a larger picture.
+
+Diving into the first part then, at a high-level this is split into three phases.
+
+#### Phase 1: Using RHEL/CentOS Stream-only bootimages
```

Comment:
Isn't there a "Phase 0" we need to link to here of shipping the result of https://github.com/openshift/os/pull/1445 in the release image? Is that already happening and I missed it?

```
$ oc adm release info registry.ci.openshift.org/ocp/release:4.17.0-0.nightly-2024-06-07-045541 |grep rhel
  rhel-coreos                                    sha256:7ab7b8bdb9f6ddde8d3f860324cacbb2e2ec7438b4228aadbf6fcb71dc1aa307
  rhel-coreos-extensions                         sha256:9af9706fa436bfc95940ef165c7a90302ce8907e5fcdb59e6cbdf542d7ba2d47
```

I expected to see something there like `rhel-coreos-base` or something that we can clearly say that the bootimages are instantiated from.

Reply:
Because in practice it *is* there, it's just not explicit.  And if the bootimages are deriving from it and not the rhel-coreos one I just think it'd be generally useful to be able to directly see it.

Mainly just for dev/support cases, there wouldn't be a customer facing case for consuming it.

Reply:
Hmm, not sure that's worth adding to the release payload for. Is there precedence for shipping images in the release payload that are only useful for support cases?

Reply:
Marking as resolved for now to clean up threads. Feel free to un-resolve and add more information.